djmoney/models/fields.py

```diff
@@ -196,25 +196,34 @@ def __init__(
         Field.creation_counter += 1
 
     def setup_default(self, default, default_currency, nullable):
-        if isinstance(default, (str, bytes)):
+        if default in (None, NOT_PROVIDED) or isinstance(default, Money):
+            return default
+        elif isinstance(default, (str, bytes)):
             try:
                 # handle scenario where default is formatted like:
                 # 'amount currency-code'
-                amount, currency = default.split(" ")
+                amount, currency = (default.decode() if isinstance(default, bytes) else default).split(" ", 1)
             except ValueError:
                 # value error would be risen if the default is
                 # without the currency part, i.e
                 # 'amount'
-                amount = default
+                amount = default.decode() if isinstance(default, bytes) else default
                 currency = default_currency
-            default = Money(Decimal(amount), Currency(code=currency))
+
+            amount = Decimal(amount)
         elif isinstance(default, (float, Decimal, int)):
-            default = Money(default, default_currency)
+            amount, currency = default, default_currency
         elif isinstance(default, OldMoney):
-            default = Money(default.amount, default.currency)
-        if default is not None and default is not NOT_PROVIDED and not isinstance(default, Money):
-            raise ValueError("default value must be an instance of Money, is: %s" % default)
-        return default
+            amount, currency = default.amount, default.currency
+        else:
+            raise ValueError(f"default value must be an instance of Money, is: {default}")
+
+        assert currency is not None, (
+            "Default currency can not be `None` when default is of a value that doesn't include a currency. Either"
+            " provide a default containing a currency value or configure a currency default via setting"
+            " `default_currency=` or `DEFAULT_CURRENCY`(global)"
+        )
```

Comment:
Is this part reachable in tests? And should it perhaps also figure as a ValueError?

Reply:
Yeah, it's reachable. It's covered by these tests:

https://github.com/django-money/django-money/blob/6dcdbeb685731ebbcf2d38638b3f6f87e21e6f1e/tests/test_models.py#L316-L320

My idea behind raising an `AssertionError` was to signal that it's an exception that's not intended to be caught by the caller.

Reply:
> My idea behind raising an AssertionError was to signal that it's an exception that's not intended to be caught by the caller.

While thinking twice about that, it'd be quite weird for someone to actually catch that exception anyways:

```python
class MyModel(models.Model):
    try:
        money = MoneyField(...)
    except ValueError:
        # ??
```

If you'd prefer a `ValueError` there, I'm fine with changing it.

Reply:
I understand your great explanation and rationale -- and no, this is not at all important to entertain going deeper into!

Reply:
Thanks for explaining!